libraries/AP_CANManager/AP_CANManager.cpp

```diff
@@ -424,13 +435,102 @@ void AP_CANManager::handle_can_frame(const mavlink_message_t &msg) const
     hal.can[p.bus]->send(frame, AP_HAL::native_micros64() + timeout_us, AP_HAL::CANIface::IsMAVCAN);
 }
 
+/*
+  handle a CAN_FILTER_MODIFY packet
+ */
+void AP_CANManager::handle_can_filter_modify(const mavlink_message_t &msg)
```

Comment:
I feel like linked list would be a better solution here rather than reallocating the array at every modification.

Reply:
I think it would make it much slower (no sorting), would take a lot more memory and would be a lot more code.
the linear array is what allows for the bisection search, which scales really well and is compact

Reply:
my worry is the approach used can lead to heap fragmentation. We should at least handle duplicate calls to same filtering config.

Reply:
I'm sure duplicate calls won't cause an issue (in fact gui_tool calls replace at 1Hz now, with no issue)
you'll just re-use the memory on the next call
the length of these lists is small, at the moment it is 11, once we have a UI for selecting messages it may grown to 30 or 40. In either case the memory usage for a linear list is tiny, and linked list would be a lot slower and use more memory